test/package_list_files_test.dart

```diff
@@ -68,14 +60,60 @@ void main() {
 
     createEntrypoint();
 
+    expect(entrypoint!.root.listFiles(), {
+      p.join(root, 'pubspec.yaml'),
+      p.join(root, 'file1.txt'),
+      p.join(root, 'file2.txt'),
+      p.join(root, 'subdir', 'a', 'file'),
+      p.join(root, 'subdir', 'symlink', 'file'),
+    });
+  });
+
+  test('not throws on ignored directory symlinks', () async {
```

Comment:
Don't we need a test case where we ignore a broken directory symlink?

And a test where we fail on a broken directory symlink?